auth/src/main/java/org/openedx/auth/presentation/signup/SignUpViewModel.kt

```diff
@@ -69,35 +74,75 @@ class SignUpViewModel(
         _uiState.update { it.copy(isLoading = true) }
         viewModelScope.launch {
             try {
-                val allFields = interactor.getRegistrationFields()
-                _uiState.update { state ->
-                    state.copy(
-                        allFields = allFields,
-                        isLoading = false,
-                    )
-                }
+                updateFields(interactor.getRegistrationFields())
             } catch (e: Exception) {
                 if (e.isInternetError()) {
                     _uiMessage.emit(
                         UIMessage.SnackBarMessage(
-                            resourceManager.getString(R.string.core_error_no_connection)
+                            resourceManager.getString(coreR.string.core_error_no_connection)
                         )
                     )
                 } else {
                     _uiMessage.emit(
                         UIMessage.SnackBarMessage(
-                            resourceManager.getString(R.string.core_error_unknown_error)
+                            resourceManager.getString(coreR.string.core_error_unknown_error)
                         )
                     )
                 }
+            } finally {
+                _uiState.update { state ->
+                    state.copy(isLoading = false)
+                }
             }
         }
     }
 
+    private fun updateFields(allFields: List<RegistrationField>) {
+        val mutableAllFields = allFields.toMutableList()
+        val requiredFields = mutableListOf<RegistrationField>()
+        val optionalFields = mutableListOf<RegistrationField>()
+        val agreementFields = mutableListOf<RegistrationField>()
+        val agreementText = agreementProvider.getAgreement(isSignIn = false)
+        if (agreementText != null) {
+            val agreement = RegistrationField(
+                name = ApiConstants.RegistrationFields.HONOR_CODE,
+                label = agreementText,
+                type = RegistrationFieldType.PLAINTEXT,
+                placeholder = "",
+                instructions = "",
+                exposed = false,
+                required = false,
+                restrictions = RegistrationField.Restrictions(),
+                options = emptyList(),
+                errorInstructions = ""
+            )
+            val honourCode = allFields.find { it.name == ApiConstants.RegistrationFields.HONOR_CODE }
+            val marketingEmails = allFields.find { it.name == ApiConstants.RegistrationFields.MARKETING_EMAILS }
```

Comment:
why we are making Marketing emails part of eula?

Reply:
During discussion on one of Technical syncs about: https://github.com/orgs/openedx/projects/56/views/7?filterQuery=eula&pane=issue&itemId=49719496 we decided to implement EULA and marketing emails